cmake/Platform/Linux/Packaging_Snapcraft.cmake

```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+include("${LY_ROOT_FOLDER}/cmake/Version.cmake")
+
+# Copy snapcraft file
+configure_file("${LY_ROOT_FOLDER}/cmake/Platform/Linux/Packaging/snapcraft.yaml.in"
+    "${CPACK_TEMPORARY_DIRECTORY}/snapcraft.yaml"
+)
+
+# build snap
+execute_process (COMMAND snapcraft --verbose
+                 WORKING_DIRECTORY ${CPACK_TEMPORARY_DIRECTORY})
+
+set(CPACK_EXTERNAL_BUILT_PACKAGES "O3DE_${O3DE_INSTALL_VERSION_STRING}_amd64.snap")
```

Comment:
nit: missing newline at end of file

Reply:
When we make a release snap, we may want to use `O3DE_INSTALL_DISPLAY_VERSION_STRING` which will be in the format XX.YY